frontend-web/src/utils/withAuth.ts

```diff
@@ -0,0 +1,19 @@
+export type User = {
+  id: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+};
+
+export async function withAuth(): Promise<User | null> {
+  try {
+    const res = await fetch('/api/auth/me');
+    if (!res.ok) {
+      return null;
+    }
+
+    return await res.json();
```

Comment:
si res.json() renvoie les données de l'user alors

```suggestion
    return { user: await res.json() };
```

et frontend-web/src/app/components/ui/AccountWidget.tsx, tu peux modifier pour récupérer directment l'objet déconstruit et l'envoyer en props